packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -82,11 +82,15 @@ def get_measurement_service_stub_and_version(
         else:
             raise
     channel = channel_pool.get_channel(service_location.insecure_address)
-    version = service_info.versions[0] if service_info.versions else ""
+
+    if not service_info.versions:
```

Comment:
The old DiscoveryService doesn't support ResolveServiceWithInformation, so if you have an older version, I don't think you'll even get this far.

I was expecting you to add a case to the `except grpc.RpcError` block for `elif e.code() == grpc.StatusCode.UNIMPLEMENTED`.

Reply:
Thanks for mentioning, I have added the check for `grpc.StatusCode.UNIMPLEMENTED`. 

When the `ni_measurement_plugin_sdk_service` is higher than 2.0.0 i.e., latest release, the `service_info.versions[0]` will throw index out of range error. To catch that `if not service_info.versions:` check was added.

Reply:
I don't understand why that would happen. Have you tested with the previous version of InstrumentStudio installed?

That said, an extra error case that doesn't get hit is not a big deal, so I approved the PR.

Reply:
>Have you tested with the previous version of InstrumentStudio installed?

Yes, I have tested with previous version of InstrumentStudio.

Reply:
I see. Mounika's PR to add ResolveServiceWithInformation to DiscoveryService is in the releases/24C2 branch, so we previously shipped a DiscoveryService with incomplete support for this API.